third_party/blink/renderer/bindings/core/v8/record_replay_interface.cc

```diff
@@ -145,6 +145,18 @@ function assert(v, msg = "") {
   }
 }
 
+/**
+ * Custom prop iterator (RUN-1315).
+ */
+function* iterateProps(o) {
+  // NOTE: symbols cannot be otherwise iterated.
+  yield *Object.getOwnPropertySymbols(o);
+  
+  for (const k in o) {
```

Comment:
For consistency's sake, wouldn't having this second part be:
```
yield *Object.getOwnPropertyNames(o);
```
be more appropriate?

Reply:
@kannanvijayan, The original "fix" here was to actually get rid of `getOwnPropertyNames`. However, it turns out that creating the iterator itself in the `for ... in` ends up also creating the set of all property names. The new fix that's coming will forego all of this, and use a modified version of `Runtime.getProperties` exclusively instead.

Reply:
However, `getOwnPropertyNames` is still way slower than the `for ... in` solution.